src/lib/utils.ts

```diff
@@ -0,0 +1,60 @@
+export function isValidUrl(url) {
+  try {
+    // Must start with http:// or https://
+    if (!/^https?:\/\/.+/i.test(url)) return false;
+    const parsedUrl = new URL(url);
```

Comment:
we use it only in server env (controlled version), use URL.parse, without try catch than